apps/server/src/adapters/utils/parse.ts

```diff
@@ -2,9 +2,19 @@
  * @description Creates a nested object with keys from an array and assigns the `value` to the last key
  * @param {array} path - array to be nested
  * @param {string} value - value to assign
- * @returns {object | null} nested object or null if no object was created
+ * @returns {object | string | null} nested object or null if no object was created
  */
-export const objectFromPath = (path: string[], value?: unknown): object | null => {
+export const integrationPayloadFromPath = (path: string[], value?: unknown): object | string | null => {
+  if (path.length == 1) {
+    if (value === undefined) {
+      return path[0];
+    } else {
+      return { [path[0]]: value };
+    }
+  }
+  if (value === undefined) {
+    value = path.pop();
+  }
   const obj = path.reduceRight((result, key) => ({ [key]: result }), value);
   if (typeof obj === 'object') return obj;
   return null;
```

Comment:
I wonder if this can be simplified, few tips:

- remember === is what you want in most cases
- avoid mutating the passed in arguments
- leave us some clues as to what things mean, like `{ [path[0]]: value };` entails a lot of knowledge about the specifics of the object, which someone who comes here to fix a bug might not have. The tests also dont provide context into the abstraction since they show the objects already abstracted. ie, as someone coming here, it is hard to piece the information from the message

how about something like (didnt test)
```ts
export const integrationPayloadFromPath = (path: string[], value?: unknown): object | string | null => {
  if (path.length === 1) {
    const key = path.at(0);
    return value === undefined ? key : { [key]: value };
  }

  const parsedValue = value === undefined ? path.at(-1) : value;

  const obj = path.reduceRight((result, key) => ({ [key]: result }), parsedValue);

  return typeof obj === 'object' ? obj : null;
};
```

Reply:
Nice and it almost worked as is